templates/layout/parts/profile_selector.html.twig

```diff
@@ -16,19 +17,116 @@
    </div>
 </div>
 
-<a href="#" class="dropdown-item show_entity_modal">
-   <i class="fa-fw fas fa-layer-group"></i>
-   {{ session('glpiactive_entity_name')|verbatim_value|u.truncate(35, '...') }}
-</a>
+{% set target = path("front/central.php") %}
+{% if get_current_interface() == "helpdesk" %}
+   {% set target = path("front/helpdesk.public.php") %}
+{% endif %}
+
+<div class="dropdown dropstart">
+   <a href="#" class="dropdown-item dropdown-toggle show_entity_modal" data-bs-toggle="dropdown" data-bs-auto-close="outside">
+      <i class="fa-fw fas fa-layer-group"></i>
+      {{ session('glpiactive_entity_name')|verbatim_value|u.truncate(35, '...') }}
+   </a>
+   <div class="dropdown-menu p-3">
+      <h3>{{ __('Select the desired entity') }}</h3>
+
+      <div class="alert alert-info" role="alert">
+         <i class="fas fa-info-circle"></i>
+         <span class="ms-2">
+            <img src="{{ path('pics/entity_all.png') }}" alt="entity_all">
+            {{ __('to see the entity and its sub-entities') }}
+         </span>
+      </div>
+
+      <form id="entsearchform{{ rand }}">
+         <div class="input-group">
+            <input type="text" class="form-control" name="entsearchtext" id="entsearchtext{{ rand }}"
+                   placeholder="{{ __('Search entity') }}" autocomplete="off">
+            <button type="submit" class="btn btn-icon btn-primary" title="{{ __('Search') }}"
+                    data-bs-toggle="tooltip" data-bs-placement="top">
+               <i class="fas fa-search"></i>
+            </button>
+            <a class="btn btn-icon btn-outline-secondary" href="#" id="entsearchtext{{ rand }}_clear"
+               title="{{ __("Clear search") }}" data-bs-toggle="tooltip" data-bs-placement="top">
+                  <i class="fas fa-times"></i>
+            </a>
+            <a href="{{ target }}?active_entity=all" class="btn btn-secondary"
+               title="{{ __('Select all') }}" data-bs-toggle="tooltip" data-bs-placement="top">
+               <i class="fas fa-eye"></i>
+            </a>
```

Comment:
At first glance, I did not understand what this "eye" could be used for.

Is this "Select all" feature really usefull, as long as is is accessible via the icon located next to the root entity ?